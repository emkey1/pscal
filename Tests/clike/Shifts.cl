int main() {
    printf("%d\n", 1 << 2 + 1);   // shift has lower precedence than + => 1 << (2+1) = 8
    printf("%d\n", 16 >> 2 + 1);  // 16 >> (2+1) = 2
    printf("%d\n", (1 << 2) + 1); // (4) + 1 = 5
    printf("%d\n", 32 >> 1 << 2); // left-associative: (32 >> 1) << 2 = 64
    return 0;
}

