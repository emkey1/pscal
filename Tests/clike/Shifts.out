8
2
5
64
